cmake/handle_eigen.cmake

```diff
@@ -0,0 +1,75 @@
+###############################################################################
+# Option for using system Eigen or GTSAM-bundled Eigen
+### These patches only affect usage of MKL. If you want to enable MKL, you *must*
+### use our patched version of Eigen
+### See:  http://eigen.tuxfamily.org/bz/show_bug.cgi?id=704 (Householder QR MKL selection)
+###       http://eigen.tuxfamily.org/bz/show_bug.cgi?id=705 (Fix MKL LLT return code)
```

Comment:
These two are fixed in 2015, I am curious that are these still relevant for us?

Reply:
You're right. I checked it and apparently the current embedded copy of Eigen doesn't have any patch applied. 
Will remove the comments. 

There were a number of comments that looked doubtful to me. Those 100% immediately-clear wrong or outdated were removed, but I left most of them untouched in the first commit of this PR to first check if all tests passed.